tests/hooks/test_hightouch_hook.py

```diff
@@ -94,7 +94,7 @@ def test_poll_fail_dont_ignore_warning(self, requests_mock):
         hook = HightouchHook()
         with pytest.raises(AirflowException) as excinfo:
             hook.poll(sync_id=1, wait_seconds=1, timeout=5, error_on_warning=True)
-            assert "Job 1 failed to complete with status warning" in str(excinfo.value)
+        assert "Job 1 failed to complete with status warning" in str(excinfo.value)
```

Comment:
@connor-charles I think the assert belongs under the `with` block, otherwise `excinfo.value` is no longer in scope, unless i'm missing something?

Reply:
Pulling the assert out of the `with` is correct. This confused me at first as well but there's an unintuitive effect going on with the context manager. When the `AirflowException` is raised by `hook.poll` it is caught by the context manager but means that the assert statement is never run. You can confirm this by asserting False instead and seeing the test still passes: 

```
  @requests_mock.mock()
  def test_poll_fail_dont_ignore_warning(self, requests_mock):
      requests_mock.get(
          "https://test.hightouch.io/api/v2/rest/sync/1",
          json=payload_factory("warning"),
      )
      hook = HightouchHook()
      with pytest.raises(AirflowException) as excinfo:
          hook.poll(sync_id=1, wait_seconds=1, timeout=5, error_on_warning=True) # Exception raised here stops code execution in context manager
          assert False, "This isn't executed!"
```

 I have changed the indentation of each test that had this problem.

Reply:
wow! i’ve been writing pytest assert exceptions wrong for as long as i’ve been writing python. thanks for that! incredibly unintuitive lol